utils/global-state-update-gen/src/generic.rs

```diff
@@ -129,16 +130,25 @@ fn update_auction_state<T: StateReader>(
         );
 
         state.remove_withdraws(&validators_diff.removed);
-    }
 
-    // All entries in the new snapshot contain the same set of validators, just use the first entry.
-    new_snapshot
-        .values()
-        .next()
-        .expect("snapshot should have at least one entry")
-        .keys()
-        .cloned()
-        .collect()
+        if validators_diff.added.is_empty() && validators_diff.removed.is_empty() {
+            None
+        } else {
+            // All entries in the new snapshot contain the same set of validators,
+            // just use the first entry.
```

Comment:
Not added in this PR, but this comment might actually be incorrect in some cases :grimacing: Feel free to leave it as is for now, but if you want to change it, I suggest something like:

> // We need to output the validators for the next era, which are contained in the first entry in the snapshot.